Libplanet/Blockchain/BlockChain.cs

```diff
@@ -165,19 +165,51 @@ IEnumerator IEnumerable.GetEnumerator()
                 _rwlock.ExitReadLock();
             }
 
-            ImmutableHashSet<Address> requestedAddresses =
-                addresses.ToImmutableHashSet();
+            ImmutableDictionary<Address, BitArray> requestedAddresses =
+                addresses.Select(addr =>
+                    new KeyValuePair<Address, BitArray>(
+                        addr,
+                        new BitArray(addr.ToByteArray())
+                    )
+                ).ToImmutableDictionary();
+            ImmutableHashSet<Address> requestedAddressSet =
+                requestedAddresses.Keys.ToImmutableHashSet();
+
+            // If there is no mask for the offset (the store made in the older
+            // versions of Libplanet may lack mask files),
+            // make the mask to match to all possible addresses.
+            var mask = new BitArray(
+                offset is HashDigest<SHA256> hash &&
+                    Store.GetAddressesMask(hash) is Address a
+                    ? a.ToByteArray()
+                    : Enumerable.Repeat((byte)0xff, Address.Size).ToArray()
+            );
+
+            Func<BitArray, BitArray, bool> isPossibilyIn = (addr, maskPat) =>
```

Comment:
I think we can use local function.

```suggestion
            bool isPossiblyIn(BitArray addr, BitArray maskPat)
            Func<BitArray, BitArray, bool> isPossibilyIn = (addr, maskPat) =>
```

Reply:
Addressed.